pages/admin/dashboard/builder.tsx

```diff
@@ -0,0 +1,268 @@
+    EditablePreview,
+    EditableInput,
+} from "@chakra-ui/react";
+import { IoIosUndo, IoIosRedo } from "react-icons/io";
+import { IoTrash, IoDesktopOutline } from "react-icons/io5";
+import { FaMobileAlt } from "react-icons/fa";
+
+import { Page } from "types/Page";
+import {
+    BuilderLayout,
+    MarkdownEditor,
+    MarkdownRenderer,
+    ModulePreview,
+    CheckboxComp,
+} from "@components";
+
+const Builder: Page = () => {
+    const {
+        isOpen: isSidebarOpen,
+        onToggle: toggleSidebar,
+        onOpen: sidebarOpen,
+        onClose: sidebarClose,
+    } = useDisclosure();
+    const [editorText, setEditorText] = useState("Hello world!");
+    const [slideNumber, setSlide] = useState(1);
+    const [maxSlides, addSlide] = useState(1);
+    const [prevSlide, setPrevSlide] = useState(1);
+    const [slides, setSlides] = useState([editorText]);
+    const [prevButton, setPrevButton] = useState(true);
+    const [nextButton, setNextButton] = useState(true);
+    const [printButton, setPrintButton] = useState(false);
+    const [saveButton, setSaveButton] = useState(false);
+
+    const moduleName = "Untitled Module";
+
+    const handleEditableErrors = (e) => {
+        const target = e.target as HTMLInputElement;
+        if (
+            !isNaN(parseInt(target.value)) &&
+            parseInt(target.value) >= 1 &&
+            parseInt(target.value) <= maxSlides
+        ) {
+            setEditorText(slides[Number(slideNumber) - 1]);
+            setPrevSlide(slideNumber);
+        } else {
+            setSlide(prevSlide);
+        }
+    };
+
+    return (
+        <Stack spacing={0}>
+            <Stack spacing={2} p={6}>
+                <Box>
+                    <Button variant="link">{"<"} Exit Builder</Button>
+                </Box>
+                <Flex>
+                    <Heading>{moduleName}</Heading>
+                    <Spacer />
+                    <HStack spacing={2}>
+                        <Button variant="outline"> Discard </Button>
+                        <Button
+                            onClick={() => {
+                                const newSlides = [...slides];
+                                newSlides[slideNumber - 1] = editorText;
+                                setSlides(newSlides);
+                            }}
+                        >
+                            Save
+                        </Button>
+                    </HStack>
+                </Flex>
+            </Stack>
+            <Flex
+                px={2}
+                py={4}
+                alignItems="center"
+                border="1px"
+                borderColor="black"
+            >
+                <Button
+                    onClick={() => {
+                        setSlide(slideNumber + 1);
+                        addSlide(maxSlides + 1);
+                        setEditorText("");
+                    }}
+                >
+                    New Slide
+                </Button>
```

Comment:
Just a hunch but this may be where the bug I identified originates from; we're not calling setSlides

Reply:
I think this bug has more to do with the MarkdownRenderer and its behaviour when it takes on empty strings "" because it cannot split an empty string, but this is something I can add to fix.

Reply:
Should have fixed as well

Reply:
Thank you for adding the fix! Great work :)